### PR TITLE
Modify OSPP rules concerning coredumps

### PR DESCRIPTION
**Pull request by vojtapolasek**
#### Description:

- drop following rules from RHEL9 OSPP
  - coredump_disable_storage
  - coredump_disable_backtraces
- add new rule sysctl_kernel_core_pattern_empty_string which sets the pattern to empty string. This rule conflicts with sysctl_kernel_core_pattern.  At first I wanted to use just single rule and utilize XCCDF variable, but it turned out to be infeasible because of limitations in Openscap handling XCCDF variables gwhen generating Ansible fixes.
- add new rule sysctl_kernel_core_uses_pid and put it into RHEL9 OSPP
- Above two rules are closely tied together, there I used `requires` statement.

#### Rationale:

https://bugzilla.redhat.com/show_bug.cgi?id=2081688

### CODE REVIEW

**Comment by vojtapolasek** [comment]
@jan-cerny  I think that we are facing a problem. There is an XCCDF variable which has the | character inside. It seems to be OK for HTML guides and Bash profile scripts, but it causes problems when generating Ansible playbooks for profiles. It seems that Ansible does not like the unescaped |. I think the problem might be in Openscap generating the fix.
Do you agree?

**Comment by jan-cerny** [comment]
We have discussed this with @vojtapolasek and also with a whole team on 2022-08-04 and we think that the options are:
* try to workaround it in the build system based on the variables value
* or create a special rule for OSPP that won't use variables

But the proper solution seems to be to fix in OpenSCAP, so I opened a ticket with them in https://github.com/OpenSCAP/openscap/issues/1879

**Comment by vojtapolasek** [comment]
I created a new rule sysctl_kernel_core_pattern_empty_string. The PR has changed including description.@jan-cerny  I think that we are facing a problem. There is an XCCDF variable which has the | character inside. It seems to be OK for HTML guides and Bash profile scripts, but it causes problems when generating Ansible playbooks for profiles. It seems that Ansible does not like the unescaped |. I think the problem might be in Openscap generating the fix.
Do you agree?

**Comment by yuumasato** [comment]
@vojtapolasek There is an unrelated commit in fd75c2c in the PR.

File: linux_os/guide/system/permissions/restrictions/sysctl_kernel_core_uses_pid/rule.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+identifiers:
+    cce@rhel9: CCE-86003-1
+
+references:
+    ospp: FMT_SMF_EXT.1
+
+ocil_clause: 'the returned line does not have a value of 0, or a line is not returned and the need for core dumps is not documented with the Information System Security Officer (ISSO) as an operational requirement'
+
+ocil: |-
+    {{{ ocil_sysctl_option_value(sysctl="kernel.core_pattern", value=0) }}}
```
Comment (yuumasato) [inline linux_os/guide/system/permissions/restrictions/sysctl_kernel_core_uses_pid/rule.yml:30]: It should be `kernel.core_uses_pid`.

File: linux_os/guide/system/permissions/restrictions/sysctl_kernel_core_uses_pid/rule.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+severity: medium
+
+identifiers:
+    cce@rhel9: CCE-86003-1
+
+references:
+    ospp: FMT_SMF_EXT.1
+
+ocil_clause: 'the returned line does not have a value of 0, or a line is not returned and the need for core dumps is not documented with the Information System Security Officer (ISSO) as an operational requirement'
```
Comment (yuumasato) [inline linux_os/guide/system/permissions/restrictions/sysctl_kernel_core_uses_pid/rule.yml:27]: Please, add line breaks to the OCIL clause.

**Review by yuumasato — CHANGES_REQUESTED** [review]

**Comment by vojtapolasek** [comment]
@yuumasato  thanks for the review, I fixed ocils.

File: linux_os/guide/system/permissions/restrictions/sysctl_kernel_core_uses_pid/rule.yml (hunk 2 above)
Comment (yuumasato) [reply linux_os/guide/system/permissions/restrictions/sysctl_kernel_core_uses_pid/rule.yml:27]: @vojtapolasek You missed this ocil_clause, :)

**Comment by vojtapolasek** [comment]
@yuumasato  ocils should be ok now

**Review by yuumasato — APPROVED** [review]

**Pull request merged**